### PR TITLE
ACS-3229: Rules v1 REST API - Get rule definition - mapping of conditions

### PR DESCRIPTION
**Pull request by krdabrowski**
Added mapping of "conditions"

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import org.alfresco.repo.action.evaluator.IsSubTypeEvaluator;
+import org.alfresco.repo.action.evaluator.NoConditionEvaluator;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.ActionCondition;
+import org.apache.commons.collections.CollectionUtils;
+
+@Experimental
+public class SimpleCondition
+{
+    private static final String COMPARATOR_EQUALS = "equals";
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java:49]: Are you planning another PR for other comparators?

File: remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+            builder.field(IsSubTypeEvaluator.PARAM_TYPE);
+            builder.comparator(COMPARATOR_EQUALS);
+            builder.parameter(actionCondition.getParameterValues().get(IsSubTypeEvaluator.PARAM_TYPE).toString());
+            break;
+        case NoConditionEvaluator.NAME:
+        default:
+            return null;
+        }
+
+        return builder.create();
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java:145]: We're not really getting neater code from the builder here.  Maybe if we had lots of methods like:
```
...
case IsSubTypeEvaluator.NAME:
    return makeIsSubTypeCondition(actionCondition);
...
private Action makeIsSubTypeCondition(actionCondition)
{
    return SimpleCondition.builder()
            .field(IsSubTypeEvaluator.PARAM_TYPE)
            .comparator(COMPARATOR_EQUALS)
            .parameter(actionCondition.getParameterValues().get(IsSubTypeEvaluator.PARAM_TYPE).toString())
            .create();
}
```

File: remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            createActionCondition(true),
+            createActionCondition()
+        );
+
+        // when
+        final CompositeCondition compositeCondition = CompositeCondition.from(actionConditions);
+
+        assertThat(compositeCondition).is(havingCompositeConditions(2, false));
+        assertThat(compositeCondition.getCompositeConditions().get(0)).is(havingSimpleConditions(2, false));
+        assertThat(compositeCondition.getCompositeConditions().get(1)).is(havingSimpleConditions(1, true));
```
Comment (tpage-alfresco) [inline remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java:63]: This test is pretty hard to follow

File: remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java (hunk 1 above)
Comment (krdabrowski) [reply remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java:49]: Generally comparators/operators are placed in parameterValue map field `.PARAM_OPERATION`. This "equals" comparator is used only for cases like: "Has aspect: audio", or "Mimetype is 3GVideo", where `.PARAM_OPERATION` is missing in Map.

File: remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java (hunk 2 above)
Comment (krdabrowski) [reply remote-api/src/main/java/org/alfresco/rest/api/model/rules/SimpleCondition.java]: Fixed according to your suggestion. Thanks

File: remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java (hunk 3 above)
Comment (krdabrowski) [reply remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java]: Simplified the test.

Comment (tpage-alfresco) [reply remote-api/src/test/java/org/alfresco/rest/api/model/rules/CompositeConditionTest.java]: Ah - I understand what the test is doing now.

**Review by tpage-alfresco — APPROVED** [review]
LGTM

**Review by mpichura — APPROVED** [review]
LGTM
